runtime/lua/vim/lsp/protocol.lua

```diff
@@ -318,6 +318,96 @@ for k, v in pairs(constants) do
   protocol[k] = tbl
 end
 
+--Lsp methods (based on version 3.18)
+protocol.Methods = {
+  Initialized = 'initialized',
+  Exit = 'exit',
+  ShutDown = 'shutdown',
+  SetTrace = '$/setTrace',
+  LogTrace = '$/logTrace',
+  ClientRegisterCapability = 'client/registerCapability',
```

Comment:
Do not use different casing than the actual protocol. That just creates useless differences.

I guess we could use underscore for the slash; nested tables is probably too clever.

```suggestion
  client_registerCapability = 'client/registerCapability',
```